src/actions/actions.ts

```diff
@@ -590,10 +590,17 @@ class CommandInsertInInsertMode extends BaseCommand {
         vimState.cursorStartPosition = leftPosition;
       }
     } else {
-      await TextEditor.insert(char, vimState.cursorPosition);
+      const selection = vscode.window.activeTextEditor.selection;
+      if (selection.isEmpty) {
+        await TextEditor.insert(char, vimState.cursorPosition);
+      } else {
+        await TextEditor.replace(selection, char);
+      }
+
+      const cursorPosition = vscode.window.activeTextEditor.selection.end;
 
-      vimState.cursorStartPosition = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
-      vimState.cursorPosition = Position.FromVSCodePosition(vscode.window.activeTextEditor.selection.start);
+      vimState.cursorStartPosition = Position.FromVSCodePosition(cursorPosition);
```

Comment:
You're right, totally copy-pasta mistake, should be `.start`.

Actually: Is there a case where we _don't_ want the `.cursorPosition` (line below) be `.selection.end`? When would we want the cursor to be at the start of the selecting after inserting text?